### PR TITLE
Container & Helm chart

### PR DESCRIPTION
**Pull request by crswty**
Build docker container and adds a helm chart for deployment to kubernetes

I think this is ready for beta testing so am keen to merge it to start getting feedback. Check the README.md in the deployment/helm folder for more detail on how to use the chart.

I've taken a best stab at hooking this into the release process but suspect there will be some finessing once we merge to get it publishing properly. There are a couple of places where I need to change the repository from crswty/postfacto to pivotal/postfacto, i'll do that post review butt pre merge.

This addresses the following issues:
https://github.com/pivotal/postfacto/issues/191
https://github.com/pivotal/postfacto/issues/115

* [x] I have reviewed the [contributing guide](https://github.com/pivotal/postfacto/blob/master/CONTRIBUTING.md)

* [x] I have made this pull request to the `master` branch

* [x] I have run all the tests using `./test.sh`.

* [x] I have added the [copyright headers](https://github.com/pivotal/postfacto/blob/master/license-header.txt) to each new file added

* [x] I have given myself credit in the [humans.txt](https://github.com/pivotal/postfacto/blob/master/humans.txt) file (assuming I want to)

### CODE REVIEW

**Comment by crswty** [comment]
Hey @textbook could you take a look at at least the docker part of this? 

Happy to take feedback on the helm stuff as well if you have some but was expecting to probably not get a review of that part.

File: .travis.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 language: minimal
+cache: bundler
```
Comment (textbook) [inline .travis.yml]: Is this going to do anything? All the installation's done inside the container.

File: deployment/helm/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+Start with this simplified install if you just want to try out Postfacto, skip to the production install
+section if you want to run an instance for real. 
+
+1. Download the chart from the [releases page](https://github.com/pivotal/postfacto/releases)  
+
+1. Run the chart:
+    ```shell script
+    helm install postfacto <downloaded-chart.tgz>
+    ```
+    If you will be running in an environemtn that doesn't support HTTPS, you will need to turn off the automatic redirect with the `disableSSLRedirect option`
```
Comment (textbook) [inline deployment/helm/README.md]: "environment"

File: docker.sh [hunk 3]
```diff
@@ -8,4 +8,4 @@ docker run -it \
   -p 4000:4000 \
   -v "$(pwd)":/postfacto \
   -v "$(pwd)/docker_node_modules:/postfacto/web/node_modules" \
-  postfacto/dev:2.6.3-12.6.0
+  postfacto-dev:latest
```
Comment (textbook) [inline docker.sh]: Does this exist? I think `postfacto/dev` is correct: https://hub.docker.com/r/postfacto/dev

File: docker/ci-push [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+set -e
+
+TAG=$1
+REPOSITORY=crswty/postfacto
```
Comment (textbook) [inline docker/ci-push]: Needs updating

File: docker/release/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+# You should have received a copy of the GNU Affero General Public License
+#
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+FROM node:12.6.0 as front-end
+
+COPY ./web /web
+WORKDIR /web
+
+RUN npm install
```
Comment (textbook) [inline docker/release/Dockerfile]: It's quicker to `npm ci` here. Also if you copy over the `package{,-lock}.json` separately first then install, _then_ copy the rest of the files, it reduces the subsequent build times for any changes that _don't_ require dependency changes.

File: docker/release/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,67 @@
+#
```
Comment (textbook) [inline docker/release/Dockerfile]: Move to root, as discussed (and required by https://devcenter.heroku.com/articles/build-docker-images-heroku-yml#known-issues-and-limitations).

File: deployment/helm/values.yaml [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+#
+#
+# You should have received a copy of the GNU Affero General Public License
+#
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+replicaCount: 1
+
+image:
+  repository: crswty/postfacto
```
Comment (textbook) [inline deployment/helm/values.yaml]: Needs updating

File: .travis.yml (hunk 1 above)
Comment (crswty) [reply .travis.yml]: Thought it was caching gems in the local directory but can't see any evidence of why I thought that now, I'll remove.

File: docker.sh (hunk 3 above)
Comment (crswty) [reply docker.sh]: Hadn't intended to commit this, reverting.

File: docker/ci-push (hunk 4 above)
Comment (crswty) [reply docker/ci-push]: 👍

File: docker/release/Dockerfile (hunk 5 above)
Comment (crswty) [reply docker/release/Dockerfile]: Switched to `npm ci` but not sure I follow how copying the package file helps, lets do that as an optimization after the fact

**Pull request merged**